src/GameLogic/PlugIns/PeriodicTasks/BloodCastleStartPlugIn.cs

```diff
@@ -0,0 +1,146 @@
+﻿// <copyright file="BloodCastleStartPlugIn.cs" company="MUnique">
+// Licensed under the MIT License. See LICENSE file in the project root for full license information.
+// </copyright>
+
+namespace MUnique.OpenMU.GameLogic.PlugIns.PeriodicTasks;
+
+using System.Runtime.InteropServices;
+using MUnique.OpenMU.GameLogic.MiniGames;
+using MUnique.OpenMU.PlugIns;
+
+/// <summary>
+/// This plugin enables the start of the blood castle.
+/// </summary>
+[PlugIn(nameof(BloodCastleStartPlugIn), "Blood Castle event")]
+[Guid("95E68C14-AD87-4B3C-AF46-45B8F1C3BC2A")]
+public sealed class BloodCastleStartPlugIn : PeriodicTaskBasePlugIn<BloodCastleStartConfiguration, BloodCastleGameServerState>, ISupportDefaultCustomConfiguration, IPeriodicMiniGameStartPlugIn
```

Comment:
Since we have now three very similar Start-classes, can we introduce a new generic base class which implements the common code?

Reply:
Sure, let me check it out.